### PR TITLE
feat: tooltip will change the position with the change of origin in custom element

### PR DESCRIPTION
**Pull request by Mr-Eve**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #3842

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/tooltip/nz-tooltip-base.directive.ts [hunk 1]
```diff
@@ -6,6 +6,14 @@
  * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
  */
 
+/**
```
Comment (wendellhu95) [inline components/tooltip/nz-tooltip-base.directive.ts]: Duplicated license info.

File: components/tooltip/nz-tooltip-base.directive.ts [hunk 2]
```diff
@@ -116,6 +125,7 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnInit, OnDes
     protected hostView: ViewContainerRef,
     protected resolver: ComponentFactoryResolver,
     protected renderer: Renderer2,
+    protected nzTooltipScrollDirective: NzTooltipScrollDirective,
```
[the remark this answers is not in the document]
Comment (Mr-Eve) [reply components/tooltip/nz-tooltip-base.directive.ts]: I have add @Optional in `NzTooltipDirective`

**Comment by Mr-Eve** [comment]
I have written a test named `should support scoller expect for body` and it passed, the screen shows `0 failures`. But the console of Karma shows that `Unhandled Promise rejection: The code should be running in the fakeAsync zone to call this function ; Zone: ProxyZone ; Task: Promise.then ; Value: Error: The code should be running in the fakeAsync zone to call this function`.

I don't know why, who can help me?

**Comment by wendellhu95** [comment]
Hi there! Could you please rebase to the lattest commit on the master branch? We've refactored this comopnent.

**Comment by wendellhu95** [comment]
Emmmm 似乎使用 CDK 提供的 CdkScrollable 就可以了
 https://github.com/angular/components/blob/d83059b6deb021193c0964eecf75452ffaf1dd0a/src/cdk/scrolling/scroll-dispatcher.ts#L54

**Comment by Mr-Eve** [comment]
I'll rebase it when I am free. And I will find some information about CdkScrollable to make up a better resolution.

**Comment by Mr-Eve** [comment]
先关闭这个PR了, 手头已安排的事情较多, 最近都没时间. 之后有空了我会重新基于现有版本提一个PR.

**Pull request closed without merging**